### PR TITLE
WSM and Terra Integration Test Fixes

### LINKED ISSUE #78: Batch pool creation fails on Terra
Batch pool creation fails with a bad request response 400, when calling Terra WSM to create a batch pool. 

The error is caused by several issues: 
 - Type mismatch between the interval for the autoscaling settings.
 - Incorrect casing in the `ApiDynamicVNetAssignmentScope` enumeration values.
 - Incorrect type in the `provision` property.

### PR DESCRIPTION
**Pull request by giventocode**
Closes #78 

- Fixes type and mapping issues to create a valid request to the WSM endpoint.
- Added exception logging to facilitate debugging
- Added required properties to the appsettings file.

### CODE REVIEW

File: src/TesApi.Web/appsettings.json [hunk 1]
```diff
@@ -5,26 +5,33 @@
...
   "CosmosDb": {
-    "CosmosDbKey": "",
-    "CosmosDbEndpoint": "",
     "AccountName": ""
   },
   "RetryPolicyOptions": {
     "MaxRetryCount": 5,
     "ExponentialBackOffExponent": 2
-  },
-  "ContainerRegistry": {
```
Comment (MattMcL4475) [inline src/TesApi.Web/appsettings.json:26]: Just checking you really meant to remove this?  if so, you can ignore this comment

**Review by MattMcL4475 — APPROVED** [review]

File: src/TesApi.Web/Management/Clients/TerraWsmApiClient.cs [hunk 2]
```diff
@@ -68,13 +70,32 @@ public virtual async Task<ApiCreateBatchPoolResponse> CreateBatchPool(Guid works
...
+            }
+            catch (Exception ex)
+            {
+                var responseContent = "";
+                if (response is not null)
+                {
+                    responseContent = await response.Content.ReadAsStringAsync();
+                }
+
+                Logger.LogError($"Failed to create a batch pool using wsm. Response content:{responseContent}", ex);
```
Comment (BMurri) [inline src/TesApi.Web/Management/Clients/TerraWsmApiClient.cs:96]: I thought that the exception goes before the message

File: src/TesApi.Web/Startup.cs [hunk 3]
```diff
@@ -157,26 +160,36 @@ private void AddAppInsightsWithDefaultOrLookingUpTheConnectionString(IServiceCol
                 s.ConnectionString = $"InstrumentationKey={instrumentationKey}";
             });
         }
-
```
Comment (BMurri) [inline src/TesApi.Web/Startup.cs:160]: There should be a blank line between methods/code blocks

**Review by BMurri — COMMENTED** [review]
I'm concerned about the order of parameters to the LogError method. I also have the same question about configuration removals as Matt. Otherwise looks good to go.

**Review by BMurri — APPROVED** [review]

File: src/TesApi.Web/Management/Clients/TerraWsmApiClient.cs (hunk 2 above)
Comment (giventocode) [reply src/TesApi.Web/Management/Clients/TerraWsmApiClient.cs:96]: Thx. Will fix.

**Pull request merged**